mk/waf.mk

```diff
@@ -0,0 +1,106 @@
+# Makefile wrapper to allow old make commands to use waf build system
+
+SYSTYPE			:=	$(shell uname)
+
+ifneq ($(findstring CYGWIN, $(SYSTYPE)),) 
+  MK_DIR := $(shell cygpath -m ../mk)
+else
+  MK_DIR := $(patsubst %/,%,$(dir $(lastword $(MAKEFILE_LIST))))
+endif
+
+include $(MK_DIR)/environ.mk
+
+
+TARGET_BOARD=sitl
+TARGET_BUILD=$(SKETCH)
+WAF_FLAGS=
+FRAME=
+BINARY_FRAME=
+
+ifeq ($(SKETCH),ArduPlane)
+TARGET_BUILD=plane
+BINARY_NAME=arduplane
+endif
+
+ifeq ($(SKETCH),ArduCopter)
+TARGET_BUILD=copter
+BINARY_NAME=arducopter
+FRAME="quad"
+BINARY_FRAME="-quad"
+endif
+
+ifeq ($(SKETCH),APMrover2)
+TARGET_BUILD=rover
+BINARY_NAME=ardurover
+endif
+
+ifeq ($(SKETCH),AntennaTracker)
+TARGET_BUILD=antennatracker
+BINARY_NAME=antennatracker
+endif
+
+FRAMES = quad tri hexa y6 octa octa-quad heli single coax
+BOARDS = px4-v1 px4-v2 px4-v4 sitl flymaple linux vrbrain vrbrain-v40 vrbrain-v45 vrbrainv-50 vrbrain-v51 vrbrain-v52 vrubrain-v51 vrubrain-v52 vrhero-v10 erle pxf navio navio2 raspilot bbbmini minlure erlebrain2 bhat qflight pxfmini
+
+define frame_template
+$(1)-$(2) : FRAME=$(2)
+$(1)-$(2) : $(1)
+$(1)-$(2) : BINARY_FRAME=-$(2)
+$(1)-$(2)-debug : $(1)-$(2)
+$(1)-$(2)-debug : WAF_FLAGS=--debug
+$(1)-$(2)-upload : $(1)-$(2)
+$(1)-$(2)-upload : WAF_FLAGS=--upload
+endef
+
+define board_template
+$(1) : TARGET_BOARD=$(1)
+$(1) : all
+$(1)-debug : $(1)
+$(1)-debug : WAF_FLAGS=--debug
+$(1)-debug : BIN_DIR=$(1)-debug
+$(1)-upload : $(1)
+$(1)-upload : WAF_FLAGS=--upload
+endef
+
+USED_BOARDS := $(foreach board,$(BOARDS), $(findstring $(board), $(MAKECMDGOALS)))
+USED_FRAMES := $(foreach frame,$(FRAMES), $(findstring $(frame), $(MAKECMDGOALS)))
+#$(warning $(USED_BOARDS))
+#$(warning $(USED_FRAMES))
+# generate targets of the form BOARD-FRAME and BOARD-FRAME-HIL
+$(foreach board,$(USED_BOARDS),$(eval $(call board_template,$(board))))
+$(foreach board,$(USED_BOARDS),$(foreach frame,$(USED_FRAMES),$(eval $(call frame_template,$(board),$(frame)))))
+
+ifeq ($(WAF_FLAGS),--debug)
+BIN_DIR=$(TARGET_BOARD)-debug
+else
+BIN_DIR=$(TARGET_BOARD)
+endif
```

Comment:
I tried it in frame_template and it didn't work, I think due to order of evaluation